### PR TITLE
Fix JRuby's scan behavior when call scan repeatedly (#45)

### PR DESCRIPTION
**Pull request by k77ch7**
@headius please review this. I think this fixs #45 and jruby/jruby#7247 .

### CODE REVIEW

File: test/strscan/test_stringscanner.rb [hunk 1]
```diff
@@ -763,6 +763,26 @@ def test_fixed_anchor_false
     assert_equal(false, StringScanner.new("a", fixed_anchor: nil).fixed_anchor?)
     assert_equal(false, StringScanner.new("a", fixed_anchor: false).fixed_anchor?)
   end
+
+  def test_scan_AREF_repeatedly
```
[the remark this answers is not in the document]
Comment (k77ch7) [reply test/strscan/test_stringscanner.rb]: @kou Thank you for your comment. I updated method name.

**Comment by enebo** [comment]
@k77ch7 This looks good.

**Comment by k77ch7** [comment]
@enebo Thank you for reviewing.

@kou Do you have the next release date?

**Comment by kou** [comment]
Done.

**Comment by k77ch7** [comment]
I confirmed.

**Pull request merged**